test/sagemaker_tests/pytorch/training/integration/sagemaker/test_smdataparallel.py

```diff
@@ -35,11 +35,14 @@ def can_run_smdataparallel(ecr_image):
     return Version(image_framework_version) in SpecifierSet(">=1.6") and Version(
         image_cuda_version.strip("cu")) >= Version("110")
 
+
 def validate_or_skip_smdataparallel_efa(ecr_image):
-    image_cuda_version = get_cuda_version_from_tag(n_virginia_ecr_image)
-    if Version(framework_version) < Version("1.8.1") or image_cuda_version != "cu111":
+    _, image_framework_version = get_framework_and_version_from_tag(ecr_image)
+    image_cuda_version = get_cuda_version_from_tag(ecr_image)
+    if Version(image_framework_version) < Version("1.8.1") or image_cuda_version != "cu111":
```

Comment:
Please change the `image_cuda_version` comparison to be similar to the type of code in lines 35-36 of this script.